cpp/src/aztec3/circuits/rollup/components/components.hpp

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include "init.hpp"
+
+namespace aztec3::circuits::rollup::components {
+std::array<fr, 2> compute_calldata_hash(std::array<abis::PreviousRollupData<NT>, 2> previous_rollup_data);
+void assert_prev_rollups_follow_on_from_each_other(BaseOrMergeRollupPublicInputs left,
+                                                   BaseOrMergeRollupPublicInputs right);
+void assert_both_input_proofs_of_same_rollup_type(BaseOrMergeRollupPublicInputs left,
+                                                  BaseOrMergeRollupPublicInputs right);
+NT::fr assert_both_input_proofs_of_same_height_and_return(BaseOrMergeRollupPublicInputs left,
+                                                          BaseOrMergeRollupPublicInputs right);
+void assert_equal_constants(BaseOrMergeRollupPublicInputs left, BaseOrMergeRollupPublicInputs right);
+
+AggregationObject aggregate_proofs(BaseOrMergeRollupPublicInputs left, BaseOrMergeRollupPublicInputs right);
+
+template <size_t N>
+NT::fr iterate_through_tree_via_sibling_path(NT::fr leaf, NT::uint32 leafIndex, std::array<NT::fr, N> siblingPath)
```

Comment:
to prevent putting the template files in the hpp we could create a components.tpp file with the templates. Not sure if this will be messier than declaring them inside this file though

Reply:
damn kinda annoying/confusing that it is already split between cpp/hpp

Reply:
Given it's quite a small file, I'd be tempted to leave them in here. If it grows, we can try to separate, perhaps?